src/lib/cache_lib/intf.ml

```diff
@@ -9,29 +12,69 @@ module Constant = struct
   end
 end
 
+(** A [('t, 'cache_t) Cached.t] is a representation of a value
+ *  stored in a cache, where ['t] is the type of the immediate
+ *  value currently being stored, and ['cache_t] is the type
+ *  of the value stored in the underlying cache. A [Cached.t] has
+ *  semantics such that it can only be consumed once, and
+ *  will track and handle the case where it is garbage collected
+ *  before consumption. [Cached.t] values can be consumed either
+ *  by transforming the value into another [Cached.t] with a
+ *  different ['t] parameter, or by invalidating the [Cached.t],
+ *  which removes it from the underlying cache.
+ *)
 module Cached = struct
   module type S = sig
     type ('t, 'cache_t) t
 
     val phantom : 't -> ('t, _) t
```

Comment:
I think `lift` actually is a good name for this function -- you're lifting a value into the cached context, but phantom seems acceptable too